frontend/app/src/components/wrapped/WrappedDialog.vue

```diff
@@ -0,0 +1,499 @@
+<script setup lang="ts">
+import dayjs from 'dayjs';
+import BigDialog from '@/components/dialogs/BigDialog.vue';
+import RotkiLogo from '@/components/common/RotkiLogo.vue';
+import AmountDisplay from '@/components/display/amount/AmountDisplay.vue';
+import HashLink from '@/components/helper/HashLink.vue';
+import AppImage from '@/components/common/AppImage.vue';
+import { type WrapStatisticsResult, useWrapStatisticsApi } from '@/composables/api/statistics/wrap';
+import { useExternalApiKeys } from '@/composables/settings/api-keys/external';
+import { useTaskStore } from '@/store/tasks';
+import { TaskType } from '@/types/task-type';
+import { Section } from '@/types/status';
+import { useCurrencies } from '@/types/currencies';
+import WrappedCard from '@/components/wrapped/WrappedCard.vue';
+import LocationDisplay from '@/components/history/LocationDisplay.vue';
+import CounterpartyDisplay from '@/components/history/CounterpartyDisplay.vue';
+import { useSupportedChains } from '@/composables/info/chains';
+import ChainDisplay from '@/components/accounts/blockchain/ChainDisplay.vue';
+import { logger } from '@/utils/logging';
+import { sortDesc } from '@/utils/bignumbers';
+import DateTimePicker from '@/components/inputs/DateTimePicker.vue';
+import { convertFromTimestamp, convertToTimestamp } from '@/utils/date';
+import { usePremium } from '@/composables/premium';
+import ExternalLink from '@/components/helper/ExternalLink.vue';
+import { useStatusUpdater } from '@/composables/status';
+import { Routes } from '@/router/routes';
+import type { BigNumber } from '@rotki/common';
+
+const props = defineProps<{
+  display: boolean;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:display', value: boolean): void;
+  (e: 'close'): void;
+}>();
+
+const { t } = useI18n();
+const premium = usePremium();
+const { apiKey } = useExternalApiKeys(t);
+const { isTaskRunning } = useTaskStore();
+const { findCurrency } = useCurrencies();
+const { fetchWrapStatistics } = useWrapStatisticsApi();
+const { getChain } = useSupportedChains();
+
+const gnosisPayKey = computed(() => apiKey('gnosis_pay'));
+const showGnosisData = computed(() => get(premium) && get(gnosisPayKey));
+
+const loading = ref(false);
+const end = ref('');
+const start = ref('');
+const summary = ref<WrapStatisticsResult | null>(null);
```

Comment:
```suggestion
const summary = ref<WrapStatisticsResult>();
```
I will merge this as it is but can we remove the `null` here and adjust things in the follow-up?